src/app/backend/apihandler.go

```diff
@@ -18,17 +18,39 @@ import (
 	"log"
 	"net/http"
 	"strconv"
+	"strings"
 
 	restful "github.com/emicklei/go-restful"
 	client "k8s.io/kubernetes/pkg/client/unversioned"
 )
 
+func WsLogger() restful.FilterFunction {
+	return func(req *restful.Request, resp *restful.Response, chain *restful.FilterChain) {
+		reqBody := make([]byte, req.Request.ContentLength)
+		req.Request.Body.Read(reqBody)
+		remoteAddr := strings.Split(req.Request.RemoteAddr, ":")[0]
```

Comment:
Are you sure that remoteAddr always contain `:`? I'm not. You should check this and do nothing when there's no `:`.